### PR TITLE
Redshift Destination: update spec

### PR DESCRIPTION
**Pull request by VitaliiMaltsev**
## What
Updated spec to follow connector release requirements

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
none

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/destination-redshift/src/main/resources/spec.json [hunk 1]
```diff
@@ -110,10 +110,10 @@
         "maximum": 100,
         "examples": ["10"],
         "description": "Optional. Increase this if syncing tables larger than 100GB. Only relevant for COPY. Files are streamed to S3 in parts. This determines the size of each part, in MBs. As S3 has a limit of 10,000 parts per file, part size affects the table size. This is 10MB by default, resulting in a default limit of 100GB tables. Note, a larger part size will result in larger memory requirements. A rule of thumb is to multiply the part size by 10 to get the memory requirement. Modify this with care.",
```
Comment (misteryeo) [inline airbyte-integrations/connectors/destination-redshift/src/main/resources/spec.json:112]:
```suggestion
        "description": "Increase this if syncing tables larger than 100GB. Only relevant for COPY. Files are streamed to S3 in parts. This determines the size of each part, in MBs. As S3 has a limit of 10,000 parts per file, part size affects the table size. This is 10MB by default, resulting in a default limit of 100GB tables. Note: a larger part size will result in larger memory requirements. A rule of thumb is to multiply the part size by 10 to get the memory requirement. Modify this with care.",
```

File: airbyte-integrations/connectors/destination-redshift/src/main/resources/spec.json [hunk 2]
```diff
@@ -95,13 +95,13 @@
       "access_key_id": {
         "type": "string",
         "description": "The Access Key Id granting allow one to access the above S3 staging bucket. Airbyte requires Read and Write permissions to the given bucket.",
```
Comment (misteryeo) [inline airbyte-integrations/connectors/destination-redshift/src/main/resources/spec.json:97]:
```suggestion
        "description": "This ID grants access to the above S3 staging bucket. Airbyte requires Read and Write permissions to the given bucket.",
```

File: airbyte-integrations/connectors/destination-redshift/src/main/resources/spec.json [hunk 3]
```diff
@@ -23,45 +23,45 @@
...
         "type": "string",
         "description": "The directory under the S3 bucket where data will be written. If not provided, then defaults to the root directory.",
         "examples": ["data_sync/test"]
       },
       "s3_bucket_region": {
-        "title": "S3 Bucket Region",
+        "title": "S3 Bucket Region (Optional)",
         "type": "string",
         "default": "",
         "description": "The region of the S3 staging bucket to use if utilising a copy strategy.",
```
Comment (misteryeo) [inline airbyte-integrations/connectors/destination-redshift/src/main/resources/spec.json:67]:
```suggestion
        "description": "The region of the S3 staging bucket to use if utilising a COPY strategy.",
```

File: airbyte-integrations/connectors/destination-redshift/src/main/resources/spec.json [hunk 4]
```diff
@@ -110,10 +110,10 @@
...
         "examples": ["10"],
         "description": "Optional. Increase this if syncing tables larger than 100GB. Only relevant for COPY. Files are streamed to S3 in parts. This determines the size of each part, in MBs. As S3 has a limit of 10,000 parts per file, part size affects the table size. This is 10MB by default, resulting in a default limit of 100GB tables. Note, a larger part size will result in larger memory requirements. A rule of thumb is to multiply the part size by 10 to get the memory requirement. Modify this with care.",
-        "title": "Stream Part Size"
+        "title": "Stream Part Size (Optional)"
       },
       "purge_staging_data": {
-        "title": "Purge Staging Files and Tables",
+        "title": "Purge Staging Files and Tables (Optional)",
         "type": "boolean",
         "description": "Whether to delete the staging files from S3 after completing the sync. See the docs for details. Only relevant for COPY. Defaults to true.",
```
Comment (misteryeo) [inline airbyte-integrations/connectors/destination-redshift/src/main/resources/spec.json:118]: Where are the docs?

"See the docs for details."

**Review by misteryeo — CHANGES_REQUESTED** [review]
Added some suggestions and comments.

File: airbyte-integrations/connectors/destination-redshift/src/main/resources/spec.json [hunk 5]
```diff
@@ -23,48 +23,48 @@
...
+        "title": "S3 Bucket Name (Optional)",
         "type": "string",
         "description": "The name of the staging S3 bucket to use if utilising a COPY strategy. COPY is recommended for production workloads for better speed and scalability. See <a href=\"https://docs.aws.amazon.com/redshift/latest/dg/c_loading-data-best-practices.html\">AWS docs</a> for more details.",
         "examples": ["airbyte.staging"]
       },
       "s3_bucket_path": {
-        "title": "S3 Bucket Path",
+        "title": "S3 Bucket Path (Optional)",
         "type": "string",
         "description": "The directory under the S3 bucket where data will be written. If not provided, then defaults to the root directory.",
```
Comment (misteryeo) [inline airbyte-integrations/connectors/destination-redshift/src/main/resources/spec.json:60]: Are there no relevant docs to link here?

File: airbyte-integrations/connectors/destination-redshift/src/main/resources/spec.json [hunk 6]
```diff
@@ -94,28 +94,28 @@
...
       },
       "part_size": {
         "type": "integer",
         "minimum": 10,
         "maximum": 100,
         "examples": ["10"],
-        "description": "Optional. Increase this if syncing tables larger than 100GB. Only relevant for COPY. Files are streamed to S3 in parts. This determines the size of each part, in MBs. As S3 has a limit of 10,000 parts per file, part size affects the table size. This is 10MB by default, resulting in a default limit of 100GB tables. Note, a larger part size will result in larger memory requirements. A rule of thumb is to multiply the part size by 10 to get the memory requirement. Modify this with care.",
-        "title": "Stream Part Size"
+        "description": "Increase this if syncing tables larger than 100GB. Only relevant for COPY. Files are streamed to S3 in parts. This determines the size of each part, in MBs. As S3 has a limit of 10,000 parts per file, part size affects the table size. This is 10MB by default, resulting in a default limit of 100GB tables. Note: a larger part size will result in larger memory requirements. A rule of thumb is to multiply the part size by 10 to get the memory requirement. Modify this with care.",
```
Comment (misteryeo) [inline airbyte-integrations/connectors/destination-redshift/src/main/resources/spec.json:112]: Are they no docs to link here?

File: airbyte-integrations/connectors/destination-redshift/src/main/resources/spec.json [hunk 7]
```diff
@@ -94,28 +94,28 @@
...
+        "title": "Stream Part Size (Optional)"
       },
       "purge_staging_data": {
-        "title": "Purge Staging Files and Tables",
+        "title": "Purge Staging Files and Tables (Optional)",
         "type": "boolean",
-        "description": "Whether to delete the staging files from S3 after completing the sync. See the docs for details. Only relevant for COPY. Defaults to true.",
+        "description": "Whether to delete the staging files from S3 after completing the sync. See the docs for details. https://docs.airbyte.com/integrations/destinations/redshift#2a-fill-up-s3-info-for-copy-strategy. Only relevant for COPY. Defaults to true.",
```
Comment (misteryeo) [inline airbyte-integrations/connectors/destination-redshift/src/main/resources/spec.json:118]: We are we including the URL directly and not hyperlinking the text?

File: airbyte-integrations/connectors/destination-redshift/src/main/resources/spec.json [hunk 8]
```diff
@@ -23,48 +23,48 @@
...
         "description": "The directory under the S3 bucket where data will be written. If not provided, then defaults to the root directory.",
         "examples": ["data_sync/test"]
       },
       "s3_bucket_region": {
-        "title": "S3 Bucket Region",
+        "title": "S3 Bucket Region (Optional)",
         "type": "string",
         "default": "",
-        "description": "The region of the S3 staging bucket to use if utilising a copy strategy.",
+        "description": "The region of the S3 staging bucket to use if utilising a COPY strategy. See the docs for details. See the docs for details. https://docs.aws.amazon.com/AmazonS3/latest/userguide/creating-bucket.html#:~:text=In-,Region,-%2C%20choose%20the%20AWS",
```
Comment (misteryeo) [inline airbyte-integrations/connectors/destination-redshift/src/main/resources/spec.json:67]: Why is the URL not hyperlinked? And I think you've put the text in twice for 'see the docs..'?

**Review by misteryeo — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/destination-redshift/src/main/resources/spec.json (hunk 6 above)
Comment (misteryeo) [reply airbyte-integrations/connectors/destination-redshift/src/main/resources/spec.json:112]: @VitaliiMaltsev?

**Review by misteryeo — APPROVED** [review]
Just had one comment here where I wasn't seeing docs but otherwise LGTM!

**Comment by VitaliiMaltsev** [comment]
> Just had one comment here where I wasn't seeing docs but otherwise LGTM!

@misteryeo added link for part_size field

**Pull request merged**